src/java_bytecode/java_types.h

```diff
@@ -507,6 +507,20 @@ class java_generic_symbol_typet : public symbol_typet
   {
     return (generic_typest &)(add(ID_generic_types).get_sub());
   }
+
+  /// Check if this symbol has the given generic type. If yes, return its index
+  /// in the vector of generic types.
+  /// \param type The type we are looking for.
+  /// \return The index of the type in the vector of generic types.
+  optionalt<size_t> generic_type(const reference_typet &type) const
+  {
+    const size_t &index =
+      std::find(generic_types().begin(), generic_types().end(), type) -
+      generic_types().begin();
```

Comment:
NIT: I think this would be easier to read if the result of `std::find` was assigned to a temporary variable and checked against `generic_types().end()` in the `if` statement below, with the subtraction only done inside the `if` statement.